source/game/scene/GameScene.cc

```diff
@@ -11,10 +11,37 @@ namespace Scene {
 GameScene::GameScene() {
     m_heap->setName("DefaultGameSceneHeap");
     m_nextSceneId = -1;
+
+    m_totalMemoryUsed = 0;
+    EGG::ExpHeap *heap = EGG::Heap::dynamicCastToExp(m_heap);
+    ASSERT(heap);
+
+    heap->calcGroupSize(&m_groupSizeRecord);
+    for (u16 groupID = 0; groupID < m_groupSizeRecord.size(); ++groupID) {
+        m_totalMemoryUsed += m_groupSizeRecord.getGroupSize(groupID);
+    }
 }
 
 /// @addr{0x8051A3C0}
-GameScene::~GameScene() = default;
+GameScene::~GameScene() {
+    m_resources.clear();
+
+    EGG::ExpHeap *heap = EGG::Heap::dynamicCastToExp(m_heap);
+    ASSERT(heap);
+
+    heap->calcGroupSize(&m_groupSizeRecord);
+    size_t sum = 0;
+    for (u16 groupID = 0; groupID < m_groupSizeRecord.size(); ++groupID) {
+        sum += m_groupSizeRecord.getGroupSize(groupID);
+    }
+
+    if (sum > m_totalMemoryUsed) {
+#ifdef BUILD_DEBUG
```

Comment:
This if-statement does nothing if this is not defined, so why bother evaluating the if-statement condition? We should move the `ifdef` to before the `if`. Even if you say the compiler will optimize this, I still think it should be changed lol

Reply:
My initial thinking was that we would warn about memory leaks on release and give specifics in debug, but to be fair this is pretty annoying. I'll be forcing all of this logic out of release and into debug.